src/main/java/ubc/pavlab/rdp/model/ontology/OntologyTermInfo.java

```diff
@@ -0,0 +1,152 @@
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString(of = { "id" }, callSuper = true)
+@org.hibernate.annotations.Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@SuperBuilder
+public class OntologyTermInfo extends OntologyTerm implements Serializable, Comparable<OntologyTermInfo> {
+
+    /**
+     * Maximum size of
+     * TODO: gather this information from the metamodel.
+     */
+    public static final int MAX_SYNONYM_LENGTH = 255;
+
+    public static OntologyTermInfoBuilder<?, ?> builder( @NonNull Ontology ontology, @NonNull String termId ) {
+        return new OntologyTermInfo.OntologyTermInfoBuilderImpl()
+                .ontology( ontology )
+                .termId( termId );
+    }
+
+    /**
+     *
+     */
+    public static Comparator<OntologyTermInfo> getComparator() {
+        return Comparator.comparing( OntologyTermInfo::getOntology, Comparator.nullsLast( Ontology.getComparator() ) )
+                .thenComparing( OntologyTermInfo::getOrdering, Comparator.nullsLast( Comparator.naturalOrder() ) )
+                .thenComparing( OntologyTermInfo::getName, Comparator.nullsLast( Comparator.naturalOrder() ) )
+                .thenComparing( OntologyTermInfo::getTermId );
+    }
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "ontology_term_info_id")
+    private Integer id;
+
+    @ElementCollection
+    @CollectionTable(name = "ontology_term_info_alt_ids", joinColumns = { @JoinColumn(name = "ontology_term_info_id") })
+    @Column(name = "alt_id", nullable = false)
+    private final Set<String> altTermIds = new HashSet<>();
+
+    @Lob
+    @Column(columnDefinition = "TEXT")
+    private String definition;
+
+    @ElementCollection
+    @CollectionTable(name = "ontology_term_info_synonyms", joinColumns = { @JoinColumn(name = "ontology_term_info_id") })
+    @Column(name = "synonym", nullable = false)
+    private final Set<String> synonyms = new HashSet<>();
+
+    /**
+     * Indicate if the term is obsolete.
+     */
+    @JsonIgnore
+    @Column(nullable = false)
+    private boolean obsolete;
+
+    /**
+     * Indicate if the term is active.
+     */
+    @JsonIgnore
+    @Column(nullable = false)
+    private boolean active;
+
+    /**
+     * Relative order of terms when sorted as per {@link #getComparator}.
+     * <p>
+     * Note that the term category has priority.
+     */
+    @JsonIgnore
+    @Column
+    private Integer ordering;
+
+    /**
+     * Indicate if the term has an icon and should be displayed.
+     */
+    @JsonIgnore
+    @Column(nullable = false)
+    private boolean hasIcon;
+
+    /**
+     * Indicate if the term is merely a grouping of other terms, in which case it will be displayed but it won't be
+     * selectable and thus convertible into a {@link UserOntologyTerm}.
+     * <p>
+     * TODO: this can be renamed
+     */
+    @JsonIgnore
+    @Column(nullable = false)
+    private boolean isGroup;
```

Comment:
It might be best to use a configuration instead of the database for storing grouping terms since it's not exactly a thing in OBO.

Reply:
We'll be using the database for this.